### PR TITLE
v4 task and container metadata handler

### PR DESCRIPTION
**Pull request by cyastella**
### Summary

This add the initials outline for the task_metadata_handler and container_metadata_handler.
### Implementation details

This PR doesn't create a v4 metadata endpoint yet, it creates the initial task_metadata_handler and container_metadata_handler. In each file we specify the path that users used to query task and container metadata for v4.

### Testing

New tests cover the changes: 

### Licensing

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: agent/handlers/v4/container_metadata_handler.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+		}
+	}
+	return containerResponse, nil
+}
+
+// GetContainerNetworkMetadata returns the network metadata for the container
+func GetContainerNetworkMetadata(containerID string, state dockerstate.TaskEngineState) ([]Network, error) {
+	dockerContainer, ok := state.ContainerByID(containerID)
+	if !ok {
+		return nil, errors.Errorf("Unable to find container '%s'", containerID)
```
Comment (fenxiong) [inline agent/handlers/v4/container_metadata_handler.go]: same as above, lowercase

File: agent/handlers/v4/container_metadata_handler.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	}
+	// the logic here has been reused from
+	// https://github.com/aws/amazon-ecs-agent/blob/0c8913ba33965cf6ffdd6253fad422458d9346bd/agent/containermetadata/parse_metadata.go#L123
+	settings := dockerContainer.Container.GetNetworkSettings()
+	if settings == nil {
+		return nil, errors.Errorf("Unable to generate network response for container '%s'", containerID)
```
Comment (fenxiong) [inline agent/handlers/v4/container_metadata_handler.go]: same as above, lowercase

File: agent/handlers/v4/container_metadata_handler.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+		responseJSON, _ := json.Marshal(containerResponse)
+		utils.WriteJSONToResponse(w, http.StatusOK, responseJSON, utils.RequestTypeContainerMetadata)
+	}
+}
+
+// GetContainerResponse gets container response for v4 metadata
+func GetContainerResponse(containerID string, state dockerstate.TaskEngineState) (*ContainerResponse, error) {
+	containerResponse, err := NewContainerResponse(containerID, state)
+	if err != nil {
+		return nil, errors.Errorf("Unable to generate metadata for container '%s'", containerID)
```
Comment (fenxiong) [inline agent/handlers/v4/container_metadata_handler.go]: probably copy pasted but error message should start with lowercase

**Comment by fenxiong** [comment]
can you add unit tests for the changes? probably in task_server_setup_test.go, same as v3

**Comment by cyastella** [comment]
> task_server_setup_test.go



> can you add unit tests for the changes? probably in task_server_setup_test.go, same as v3

sure, will add it.

**Pull request merged**